resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -443,19 +444,28 @@ private[spark] object Client extends Logging {
       appArgs = appArgs).run()
   }
 
-  def resolveK8sMaster(rawMasterString: String): String = {
+  /**
+   * @return a String URL if specified by the master URL, or empty if the client default is
+   *         specified using k8s
+   * @throws IllegalArgumentException for master strings not k8s or starting with k8s://
+   */
+  private def resolveK8sMaster(rawMasterString: String): Option[String] = {
+    if (rawMasterString.equals("k8s")) {
+      return Option.empty
```

Comment:
Or use a case statement:

`rawMasterString match {
  case "k8s" => None
  case other if rawMasterString.startsWith("k8s://") => <handle master>
  case _ => throw new SparkException
}`

I think the `other if ...startsWith` case can use a regex to match the prefix but I'm not sure if there's Scala syntactic sugar for this.